Base/Python/slicer/__init__.py

```diff
@@ -1,4 +1,108 @@
-""" This module sets up root logging and loads the Slicer library modules into its namespace."""
+""" This module sets up root logging and loads the Slicer library modules into its namespace.
+
+
+.. py:attribute:: app
+
+  This is set to the singleton instance of ``qSlicerApplication``::
+
+    >>> slicer.app
+    qSlicerApplication (qSlicerApplication at: 0x7ffd066a07a0)
+
+  .. warning::
+
+    This attribute is set only in the context of a running Slicer application.
```

Comment:
It is not very clear what we mean by "in the context of a running Slicer application". Something like this may be better:
```suggestion
    This attribute is only set in the Python environment embedded in the Slicer main application SlicerApp-real (launched by Slicer executable). The attribute is not set in the Python environment of python-real (launched by PythonSlicer executable).
```